gptme/log2html.py

```diff
@@ -0,0 +1,123 @@
+import html
+import logging
+from pathlib import Path
+from typing import List
+
+from .message import Message
+
+logger = logging.getLogger(__name__)
+
+def export_to_html(log: List[Message], output_path: Path) -> None:
+    """
+    Export the conversation log to a self-contained HTML file.
+
+    This function generates an HTML file that includes the conversation log,
+    styled with Tailwind CSS. The HTML file is self-contained and can be
+    uploaded to a static site or included in another page.
+
+    The Content Security Policy (CSP) is set to allow necessary resources
+    while still maintaining a good level of security.
+
+    Args:
+        log (List[Message]): The conversation log to export.
+        output_path (Path): The path where the HTML file will be saved.
+
+    Returns:
+        None
+    """
+    html_content = f"""
```

Comment:
It might be better to build a self-contained bundle out of the already existing frontend, to maximize code reuse.